### PR TITLE
Remove prohibition against building Universal Binaries on macOS

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Comment by seanm** [comment]
@epourmal

**Review by byrnHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Comment by seanm** [comment]
Ideally this would be backported to earlier branches too.  Personally, I use 1.10.x and would most like to see it there (and newer).

**Comment by seanm** [comment]
@imikejackson

**Comment by byrnHDF** [comment]
Should there be a RELEASE.txt entry?

**Review by byrnHDF — COMMENTED** [review]
Also, is the user expected to know how to build a universal binary and when not to?

**Comment by seanm** [comment]
> Also, is the user expected to know how to build a universal binary and when not to?

Depending on who you mean by "user", then yes, any Mac programmer building HDF5 would know when/if he wants to build universal.  Now that Apple is selling both Intel and ARM Macs, the 'when' is usually "always".

**Comment by imikejackson** [comment]
For what it's worth I'm ok with these changes.

I also wouldn't mind seeing this back ported to the 1.10.x branch as that is what we are using in our projects.

**Review by byrnHDF — CHANGES_REQUESTED** [review]
I think we do need to have a release note entry about this.
Just something simple that dates the change and why.

**Review by qkoziol — APPROVED** [review]

**Comment by byrnHDF** [comment]
Once the change is in develop, I can make the merges to 1.12 and 1.10.
Allen

**Comment by seanm** [comment]
OK, I'll amend this with a note in the release notes...

**Comment by seanm** [comment]
Where should the note go?  In RELEASE.txt?

**Comment by byrnHDF** [comment]
Correct, newest at top of section Configuration. I would use the "New Features"  Configuration section rather then the "Bug Fixes".

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -48,6 +48,11 @@ New Features
 
     Configuration:
     -------------
+    - On macOS, Universal Binaries can now be built, allowing native execution on
+      both Intel and Apple Silicon (ARM) based Macs.
+
+      To do so, set CMAKE_OSX_ARCHITECTURES="x86_64;arm64"
+
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt:55]: One more line - initials and date, please.

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**